tests/virt_autotest/qem/live_migration_src.pm

```diff
@@ -0,0 +1,217 @@
+    } elsif (check_var('SYSTEM_ROLE', 'xen')) {
+        create_vm($vms, $guest, "fv", "x86_64");
+        create_vm($vms, $guest, "pv", "x86_64");
+    }
+}
+
+# perform live migration test and clean vm up after.
+sub migrate_test {
+    my ($vm_name, $vm_info) = @_;
+    for my $cmd (@{$vm_info->{virsh_cmds}}) {
+        #check if ip file is created. timeouts after 9 retries
+        assert_script_run("check_ip_file(){ for i in \$(seq 1 9);do if [ -f $images_path/guestip/$vm_name ]; then sleep 10; return; else sleep 10; fi; done; false; }; check_ip_file");
+        $vm_info->{ip} = script_output("cat $images_path/guestip/$vm_name");
+        if ("$cmd" eq "xl migrate") {
+            record_info("$cmd $vm_name $dst_ip");
+            assert_script_run("$cmd $vm_name $dst_ip");
+        } else {
+            record_info("$cmd $vm_name $vm_info->{dsturi}");
+            assert_script_run("$cmd $vm_name $vm_info->{dsturi}");
+        }
+        # visit web server running in vm check it's content
+        validate_script_output("curl -s http://$vm_info->{ip}/guest", sub { m/This is guest $vm_name/ });
+        #test vm disk write and read
+        assert_script_run("echo $vm_name > /tmp/test_vm_write");
+        assert_script_run("curl -s -k -T /tmp/test_vm_write -u bernhard:$testapi::password scp://$vm_info->{ip}/~/test_vm_write");
+        validate_script_output("curl -s -k -u bernhard:$testapi::password scp://$vm_info->{ip}/~/test_vm_write", sub { m/$vm_name/ });
+        assert_script_run("rm $images_path/guestip/$vm_name");
+        #shutdown migrated vm
+        if ($cmd eq "xl migrate") {
+            assert_script_run("ssh root\@$dst_ip xl shutdown $vm_name");
+        } else {
+            assert_script_run("ssh root\@$dst_ip virsh shutdown $vm_name");
+        }
+        #wait for shutdown complete
+        sleep 15;
+        #start again for next migration test
+        assert_script_run("virsh start $vm_name") if ($cmd ne $vm_info->{virsh_cmds}->[-1]);
+    }
+}
+
+sub cleanup {
+    my ($vm_name, $vm_info) = @_;
+    assert_script_run("virsh dumpxml $vm_name > $images_path/vm-configs/$vm_name.xml") unless ($vm_info->{reuse});
+    assert_script_run("virsh undefine $vm_name");
+}
+
+sub run {
+    select_console 'root-ssh';
+    $dst_ip = get_job_autoinst_vars(get_parents->[0])->{SUT_IP};
+    record_info("DST: $dst_ip");
+    record_info("SUT: $sut_ip");
+    assert_script_run("mount $sut_ip:/tmp/vm_images $images_path") unless get_var("NFS_PATH");
+    assert_script_run("rm -rf $images_path/guestip; mkdir $images_path/guestip");
+    assert_script_run("mkdir -p $images_path/vm-configs");
+    my $guests = get_var_array('GUESTS');
+    #start all guests
+    for my $guest (@{$guests}) { start_guest($guest, $vms); }
+
+    assert_script_run('if [ -d /root/.ssh ]; then rm -rf /root/.ssh; fi');
+    assert_script_run('ssh-keygen -t rsa -q -f "/root/.ssh/id_rsa" -N ""');
+
+    mutex_wait 'dst_ready';
```

Comment:
Perhaps you want to consider to put lines 159-167 into a subroutine like `sub prepare_host` or something similar?

Reply:
line164 is not host preparation. It starts creating VMs. 

Moreover, they are executed just once in this test. There is no reuse need.  I would prefer just  keep them in run() subroutine.